app/feature/insurances/src/test/kotlin/com/hedvig/android/feature/insurances/insurance/presentation/InsurancePresenterTest.kt

```diff
@@ -0,0 +1,263 @@
+      id = "contractId#2",
+      displayName = "displayName#2",
+      statusPills = listOf("statuspill#1"),
+      detailPills = listOf("detailpill#1"),
+      isTerminated = false,
+      typeOfContract = TypeOfContract.SE_APARTMENT_RENT,
+    ),
+  )
+  private val terminatedContracts: List<InsuranceContract> = listOf(
+    InsuranceContract(
+      id = "contractId#3",
+      displayName = "displayName#3",
+      statusPills = listOf("statuspill#1"),
+      detailPills = listOf("detailpill#1"),
+      isTerminated = true,
+      typeOfContract = TypeOfContract.SE_APARTMENT_RENT,
+    ),
+    InsuranceContract(
+      id = "contractId#4",
+      displayName = "displayName#4",
+      statusPills = listOf("statuspill#1"),
+      detailPills = listOf("detailpill#1"),
+      isTerminated = true,
+      typeOfContract = TypeOfContract.SE_APARTMENT_RENT,
+    ),
+  )
+  private val validCrossSells: List<CrossSalesQuery.Data.CurrentMember.CrossSell> = listOf(
+    CrossSalesQuery.Data.CurrentMember.CrossSell(
+      id = "crossSellId",
+      title = "crossSellTitle",
+      description = "crossSellDescription",
+      storeUrl = "",
+    ),
+  )
+
+  @Test
+  fun `on launch loading state is true, followed by the loaded state`() = runTest {
+    val getInsuranceContractsUseCase = FakeGetInsuranceContractsUseCase()
+    val getCrossSellsUseCase = FakeGetCrossSellsUseCase()
+    val presenter: InsurancePresenter = InsurancePresenter(
+      getInsuranceContractsUseCase,
+      getCrossSellsUseCase,
+      FakeCrossSellCardNotificationBadgeService(),
+    )
+    presenter.test(InsuranceUiState.InitialState) {
+      awaitItem().also { uiState ->
+        assertThat(uiState).isEqualTo(InsuranceUiState.InitialState)
+        assertThat(uiState.loading).isTrue()
+      }
+
+      getInsuranceContractsUseCase.contracts.add(validContracts)
+      getCrossSellsUseCase.crossSells.add(validCrossSells)
+      awaitItem().also { uiState ->
+        assertAll {
+          assertThat(uiState.hasError).isFalse()
+          assertThat(uiState.loading).isFalse()
+          assertThat(uiState.quantityOfCancelledInsurances)
+            .isEqualTo(validContracts.count(InsuranceContract::isTerminated))
+          assertThat(uiState.insuranceCards.map { it.contractId }).containsSubList(validContracts.map { it.id })
+          assertThat(uiState.crossSells.map { it.id }).containsSubList(validCrossSells.map { it.id })
+          assertThat(uiState.showNotificationBadge)
+        }
+      }
+    }
+  }
+
+  @Test
+  fun `when cross sells fail to load, we get an error state`() = runTest {
+    val getInsuranceContractsUseCase = FakeGetInsuranceContractsUseCase()
+    val getCrossSellsUseCase = FakeGetCrossSellsUseCase()
+    val presenter = InsurancePresenter(
+      getInsuranceContractsUseCase,
+      getCrossSellsUseCase,
+      FakeCrossSellCardNotificationBadgeService(),
+    )
+    presenter.test(InsuranceUiState.InitialState) {
+      skipItems(1)
```

Comment:
We've got a test which tests the initial state already, so for the rest of the tests since we know we're testing UI state, we know the first emission is always the initial state, so we can just skip the first one.
This is the same as doing `awaitItem()` and then not doing anything with the response.